extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -41,9 +29,13 @@ public function index() {
 
 		$this->wg->Out->setPageTitle( wfMessage( 'discussions-pagetitle' )->escaped() );
 
-		if ( $this->request->wasPosted() ) {
-			$this->activateDiscussions();
-			WikiFactory::setVarByName( 'wgEnableDiscussions', $this->siteId, true );
+		if ( $this->request->wasPosted() && $this->assertValidPostRequest() ) {
+			if ( !SpecialDiscussionsHelper::activateDiscussions(
+					$this->siteId,
+					$this->app->wg->ContLang->getCode(),
+					$this->siteName ) ) {
+				throw new ErrorPageError( 'unknown-error', 'discussions-activate-error' );
```

Comment:
I'd disagree on this point - I think it makes more sense for it to return a bool (`TRUE` when discussions was successfully activated, `FALSE` otherwise) in this case when it's a static method.